samples/graph-activity-feed-broadcast/csharp/ActivityFeedBroadcast/Controllers/HomeController.cs

```diff
@@ -0,0 +1,229 @@
+using ActivityFeedBroadcast.Model;
+using System.Net.Http.Headers;
+using Polly;
+using System.Net;
+using Newtonsoft.Json;
+using System.Text;
+
+namespace ActivityFeedBroadcast.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly ConcurrentDictionary<string, List<BroadcastInfo>> _taskList;
+
+        public HomeController(
+            IConfiguration configuration,
+            IHttpClientFactory httpClientFactory,
+            IHttpContextAccessor httpContextAccessor,
+            ConcurrentDictionary<string, List<BroadcastInfo>> taskList)
+        {
+            _configuration = configuration;
+            _httpClientFactory = httpClientFactory;
+            _httpContextAccessor = httpContextAccessor;
+            _taskList = taskList;
+        }
+
+        // Get broadcast page.
+        [Route("broadcast")]
+        public ActionResult Broadcast()
+        {
+            return View("Index");
+        }
+
+        // Get broadcast message Id.
+        [HttpGet]
+        [Route("MessageDetails")]
+        public ActionResult GetRequestByID(string taskId)
+        {
+            var currentTaskList = new List<BroadcastInfo>();
+            _taskList.TryGetValue("taskList", out currentTaskList);
+
+            if (currentTaskList == null)
+            {
+                ViewBag.Message = "No record found";
+            }
+            else
+            {
+                var request = currentTaskList.FirstOrDefault(p => p.taskId.ToString() == taskId);
+                ViewBag.TaskDetails = request;
+            }
+            return View("Message");
+        }
+
+        // Send notification toorganisation.
+        [HttpPost]
+        [Route("SendNotificationToOrganisation")]
+        public async Task<ActionResult> SendNotificationToOrganisation(BroadcastInfo taskInfo)
+        {
+            try
+            {
+                var currentTaskList = new List<BroadcastInfo>();
+                List<BroadcastInfo> taskList = new List<BroadcastInfo>();
+                _taskList.TryGetValue("taskList", out currentTaskList);
+
+                taskInfo.taskId = Guid.NewGuid();
+
+                if (currentTaskList == null)
+                {
+                    taskList.Add(taskInfo);
+                    _taskList.AddOrUpdate("taskList", taskList, (key, newValue) => taskList);
+                    ViewBag.TaskList = taskList;
+                }
+                else
+                {
+                    currentTaskList.Add(taskInfo);
+                    _taskList.AddOrUpdate("taskList", currentTaskList, (key, newValue) => currentTaskList);
+                    ViewBag.TaskList = currentTaskList;
+                }
+
+                var graphClient = SimpleGraphClient.GetGraphClient(taskInfo.access_token);
```

Comment:
Do  we need separate variables for graphClient? if yes, we can add some comment here.

Reply:
Added comments